services/filter/src/filter/routes/filter.py

```diff
@@ -0,0 +1,240 @@
+                    ):
+                        raise MissingRequiredParameterError(
+                            "Parameters 'dataset', 'config', 'split' and 'where' are required"
+                        )
+                    # TODO: validate where
+                    offset = int(request.query_params.get("offset", 0))
+                    if offset < 0:
+                        raise InvalidParameterError(message="Parameter 'offset' must be positive")
+                    length = int(request.query_params.get("length", MAX_ROWS))
+                    if length < 0:
+                        raise InvalidParameterError("Parameter 'length' must be positive")
+                    elif length > MAX_ROWS:
+                        raise InvalidParameterError(f"Parameter 'length' must not be bigger than {MAX_ROWS}")
+                    logger.info(
+                        f'/filter, dataset={dataset}, config={config}, split={split}, where="{where}",'
+                        f" offset={offset}, length={length}"
+                    )
+                with StepProfiler(method="filter_endpoint", step="check authentication"):
+                    # If auth_check fails, it will raise an exception that will be caught below
+                    auth_check(
+                        dataset=dataset,
+                        external_auth_url=external_auth_url,
+                        request=request,
+                        hf_jwt_public_keys=hf_jwt_public_keys,
+                        hf_jwt_algorithm=hf_jwt_algorithm,
+                        hf_timeout_seconds=hf_timeout_seconds,
+                    )
+                with StepProfiler(method="filter_endpoint", step="get parquet file metadata items from cache"):
+                    parquet_file_metadata_items, revision = get_config_parquet_metadata_from_cache(
+                        dataset=dataset, config=config, split=split, processing_graph=processing_graph
+                    )
+                with StepProfiler(method="filter_endpoint", step="get parquet file urls"):
+                    parquet_file_urls = [item["url"] for item in parquet_file_metadata_items]
+                with StepProfiler(method="filter_endpoint", step="get features"):
+                    features = get_features_from_parquet_file_metadata(
+                        parquet_file_metadata_item=parquet_file_metadata_items[0],
+                        parquet_metadata_directory=parquet_metadata_directory,
+                    )
+                with StepProfiler(method="filter_endpoint", step="get supported and unsupported columns"):
+                    supported_columns, unsupported_columns = get_supported_unsupported_columns(
+                        features,
+                        unsupported_features=UNSUPPORTED_FEATURES,
+                    )
+                with StepProfiler(method="filter_endpoint", step="execute filter query"):
+                    num_rows_total, pa_table = execute_filter_query(
```

Comment:
is pa_table copied in the memory?

Reply:
I aligned with the ocde in /search:
https://github.com/huggingface/datasets-server/blob/88257755b2b8ab5a16540f98b621aab826f1dfe5/services/search/src/search/routes/search.py#L268

And yes, I think the pa_table is created in memory. It only has `MAX_ROWS = 100` though... Anyway, sooner or later we need the data in memory to render it on the viewer. See code in `to_rows_list`:
https://github.com/huggingface/datasets-server/blob/88257755b2b8ab5a16540f98b621aab826f1dfe5/libs/libapi/src/libapi/utils.py#L212

Reply:
OK, you're right, it's small